File: WebApp.Core/Entities/Compensation/PaymentTable.cs
namespace WebApp.Core.Entities.Compensation
{
    using System;
    using System.Collections.Generic;

    public class PaymentTable : IEntityBase
    {
        public DateTime? FinalDate { get; set; }

        public int Id { get; set; }

        public int TableType { get; set; }

        public int ThresholdType { get; set; }

        public int? PercentageIndicatorType { get; set; }

        public int Version { get; set; }

        public string Name { get; set; }

        public DateTime StartDate { get; set; }

        public List<PaymentTableIndicator> Indicators { get; set; }

        public List<PaymentTableHierarchy> Hierarchies { get; set; }
    }
}

File: BaseWebApp/ClientApp/src/app/core/models/compensation/PaymentTable.ts
import { IPaymentTableIndicator, IPaymentTableHierarcy } from '@app/core';

export interface IPaymentTable {
    id: number;
    version: number;
    name: string;
    tableType: number;
    thresholdType: number;
    percentageIndicatorType?: number;
    startDate: Date;
    finalDate?: Date;
    activeStatus: boolean;
    indicators: IPaymentTableIndicator[];
    hierarchies: IPaymentTableHierarcy[];
}
